build/llhttp.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+const { join, resolve } = require('path')
+const { copyFileSync, rmSync } = require('fs')
+const { execSync } = require('child_process')
+const TMP = require('os').tmpdir()
+
+const REPO = 'git@github.com:dnlup/llhttp.git'
+const CHECKOUT = 'undici_wasm'
```

Comment:
Should we checkout a specific commit hash?

Reply:
Can we include the stuff from that branch here and clone the master branch?

Reply:
> Can we include the stuff from that branch here and clone the master branch?

So, the changes required to compile a usable `wasm` binary are in:

* `src/native/api.c`
* `src/native/api.h`

which are also used to build the `c` code for the parser. In light of this, I formed the opinion that it is responsibility of `llhttp` providing a usable `wasm` build. This is done as a concept in https://github.com/dnlup/llhttp/tree/undici_wasm. All the build logic is there (docker +  helper script). All we should do here imho is download the repo, build and copy the binary/assets required. With the - limited - knowledge I currently have, I think this is the cleanest road. That would require us to merge these changes upstream or use a custom branch/repo.

So my answer I guess is, we can, maybe it would be messier?

> Should we checkout a specific tag or commit hash?

Sure, this is just a concept, we could use a tag or a specific commit.

Reply:
Sounds good to me. Let’s do it like you propose.